doc/design/model_zoo_pai.md

```diff
@@ -0,0 +1,111 @@
+    
+    ```
+
+    When a user runs the above command, the CLI will submit the task to PAI then print a logview URL and an instance ID to stdout, after which the user can safely terminate the CLI. The user can check the logview to get the running status of the task or run:
+    
+    ```bash
+    $ odpscmd -e "wait instanceID"
+    ```
+    
+    to attach to the running task.
+
+1. Use the MaxCompute Python Package [`pyodps`](https://pyodps.readthedocs.io/en/latest/):
+
+    ```python
+    from odps import ODPS
+    
+    conn = ODPS(...)
+    inst = conn.run_xflow('tensorflow', parameters={'script':      'file:///my/training/script.py',
+                   	                                'other_param': 'other_param_value' ... })
+    print(instance.get_logview_address())
+    ```
+    
+    This code snippet asynchronously submits a task to PAI and prints the logview URL to stdout.
+
+**Note** that the code snippets here are only for demonstration. In practice, there may be multiple python scripts in a PAI program. If so, we can pack all the python scripts/dependencies of the PAI program into a tarball that can be passed to the `-Dscript` option. PAI will automatically unpack the tarball. And PAI has another command line option to specify which python script is the entry of the program. For details, please refer to the user guide of [odpscmd](https://www.alibabacloud.com/help/doc-detail/27971.htm) and [pyodps](https://pyodps.readthedocs.io/en/latest/).
+
+## The Design
+
+### Versioning and Releasing
+
+Versioning and releasing in PAI model zoo is the same as what's described in [Model Zoo Design](model_zoo.md#Versioning-and-Releasing). The only requirement is that the base docker image of the SQLFlow model zoo should incorporate both `odpscmd`(which is already in place) and `pyodps`.  We propose to use `pyodps` at the moment because it is easier to be integrated into the new workflow framework that is in progress.
+
+### Submitter Programs of PAI Model Zoo
+
+Currently, each deployment of SQLFlow has been configured to use only one submitter. So we assume that all the tasks of the deployment of SQLFlow on PAI will be submitted to PAI.
+
+When a user submits a SELECT statement as described in [Model Zoo Background](model_zoo.md#Background), SQLFlow should take the following actions:
+
+1. SQLFlow Checks whether the entity after `TO TRAIN/PREDICT/ANALYZE` is from a SQLFlow model zoo. For example, a plain `DNNClassier` implies that the model is a premade estimator, and `"models.sqlflow.org/sqlflow/my_awesome_model"` implies that the model is from model zoo "models.sqlflow.org". The actual mechanism may be more complicated and is still under progress.
+1. Case A, the model **is not** from a model zoo:
+    - The SQLFlow server generates a submitter program with PAI-required command line options.
+	- The SQLFlow server uses `odpscmd` or `pyodps` to submit the program to PAI as [described above](#Background). 
```

Comment:
Should be couler will have a module like `couler.pai` to use `pyodps` to submit jobs to PAI?

Reply:
I think so, submitting a PAI program is sort of complicated.